src/components/schedule/ClassCard.tsx

```
@@ -140,42 +158,54 @@ export const ClassCard: React.FC<ClassCardProps> = ({
+            <button
+              className={`px-6 md:px-8 py-2 md:py-3 rounded-xl text-sm md:text-lg font-semibold transition-all duration-300 ${
+                classInfo.enrolled >= classInfo.room.capacity
+                  ? "bg-gray-400 text-white cursor-not-allowed"
+                  : loadingAllocation === classInfo.id
+                  ? "bg-gradient-to-r from-green-600/80 to-teal-600/80 text-white cursor-wait"
+                  : "bg-gradient-to-r from-green-600 to-teal-600 text-white hover:shadow-lg transform hover:scale-[1.02]"
+              }`}
+              disabled={classInfo.enrolled >= classInfo.room.capacity || loadingAllocation === classInfo.id}
+              onClick={onBookClass}
+            >
+              {loadingAllocation === classInfo.id ? (
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)